File: AutomaticTimetableGeneratorSystem-main/timetable-frontend/src/services/holidayService.js
import api from "./api";

const BASE_URL = "/holidays";

const holidayService = {

    getAllHolidays: async () => {

        const response =
            await api.get(BASE_URL);

        return response.data;

    },


    getHolidayById: async (id) => {

        const response =
            await api.get(
                `${BASE_URL}/${id}`
            );

        return response.data;

    },


    createHoliday: async (holiday) => {

        const response =
            await api.post(
                BASE_URL,
                holiday
            );

        return response.data;

    },


    updateHoliday: async (
        id,
        holiday
    ) => {

        const response =
            await api.put(
                `${BASE_URL}/${id}`,
                holiday
            );

        return response.data;

    },


    deleteHoliday: async (
        id
    ) => {

        const response =
            await api.delete(
                `${BASE_URL}/${id}`
            );

        return response.data;

    },


    getHolidayByDate: async (
        holidayDate
    ) => {

        const response =
            await api.get(
                `${BASE_URL}/date/${holidayDate}`
            );

        return response.data;

    }

};


export default holidayService;

File: AutomaticTimetableGeneratorSystem-main/timetable-backend/src/main/java/com/timetable/service/HolidayService.java
package com.timetable.service;

import java.time.LocalDate;
import java.util.List;

import com.timetable.dto.holiday.HolidayRequest;
import com.timetable.dto.holiday.HolidayResponse;

public interface HolidayService {

    /**
     * Save Holiday
     */
    HolidayResponse saveHoliday(
            HolidayRequest request);

    /**
     * Update Holiday
     */
    HolidayResponse updateHoliday(
            Long holidayId,
            HolidayRequest request);

    /**
     * Delete Holiday
     */
    void deleteHoliday(Long holidayId);

    /**
     * Get Holiday By ID
     */
    HolidayResponse getHolidayById(
            Long holidayId);

    /**
     * Get All Holidays
     */
    List<HolidayResponse> getAllHolidays();

    /**
     * Get Holiday By Date
     */
    HolidayResponse getHolidayByDate(
            LocalDate holidayDate);

    /**
     * Check Whether Date Is Holiday
     * (Used By Timetable Generator)
     */
    boolean isHoliday(
            LocalDate holidayDate);
}